tools/clean/provider/gcp.go

```diff
@@ -100,6 +103,73 @@ func (gcp *GCP) DeletePrivateEndpoint(ctx context.Context, groupName, sAttachmen
 	return nil
 }
 
+func (gcp *GCP) DeleteOrphanPrivateEndpoints(ctx context.Context, lifetime int, region string, subnet string) error {
+	addresses := gcp.addressClient.List(ctx, &computepb.ListAddressesRequest{
+		Project: gcp.projectID,
+		Region:  region,
+	})
+	for {
+		addr, err := addresses.Next()
+		if errors.Is(err, iterator.Done) {
+			break
+		}
+		if err != nil {
+			return err
+		}
+		suffix := fmt.Sprintf("subnetworks/%s", subnet)
+		if !strings.HasSuffix(addr.GetSubnetwork(), suffix) {
+			fmt.Println(text.FgYellow.Sprintf("\tAddress %s(%s) skipped, not in %s\n",
+				addr.GetName(), addr.GetAddress(), subnet))
+			continue
+		}
+		createdAt, err := asTime(addr.GetCreationTimestamp())
+		if err != nil {
+			return err
+		}
+		if time.Since(createdAt) < time.Duration(lifetime)*time.Hour {
+			fmt.Println(text.FgYellow.Sprintf(
+				"\tAddress %s(%s) skipped once created less than %d hours ago\n",
+				addr.GetName(), addr.GetAddress(), lifetime))
+		}
+		frName, err := expectForwardingRule(addr.GetUsers())
+		if err != nil {
+			return err
```

Comment:
this is actually fine not wrapping, as the error is generated here, only once, and is specific :-)